### ISSUE TITLE
Does the repo support OFA pre-training?

### ISSUE THREAD

**Issue by zzhanghub** [OFA-Sys/OFA-Compress#1]
Awesome job! The Huggingface version OFA looks more concise. Can this framework support multi-task pre-training like this [repo](https://github.com/OFA-Sys/OFA)?
I see that this[ code file](https://github.com/OFA-Sys/OFA-Compress/blob/main/data_utils/unify_dataset.py) contains many different tasks. Could you provide more details about pre-training (Such as data preparation and submission script)?

Thank you again and look forward to your reply!

**Comment by faychu** [comment]
Thanks for your support! I just released a version of the pretrain script. Please try [pretrain.sh](https://github.com/OFA-Sys/OFA-Compress/blob/main/scripts/pretrain.sh). But I do not recommend using this framework for pre-training. We have tried pre-training under this framework before, and its performance is not as good as the original repo (fairseq version).

**Comment by zzhanghub** [comment]
> Thanks for your support! I just released a version of the pretrain script. Please try [pretrain.sh](https://github.com/OFA-Sys/OFA-Compress/blob/main/scripts/pretrain.sh). But I do not recommend using this framework for pre-training. We have tried pre-training under this framework before, and its performance is not as good as the original repo (fairseq version).
@faychu 

Thank you for your prompt reply.

Does this suboptimal performance refer to training from scratch? Have you tried to load ckpt (from fairseq version) to continue adding new data or tasks for pre-training?

By the way, will OFA continue to maintain the pre-training in fairseq version in the future, or gradually change to the Huggingface version?

Thank you!!!

**Comment by faychu** [comment]
Yes, the performance refers to training from scratch, and we haven't tried pertaining with new data from loaded ckpt. If you are interested, welcome to try it. 

In the future, the OFA pre-training will still be maintained in fairseq version, but the compression features (including distillation, pruning and quantization) will be supported by this repo.

**Comment by zzhanghub** [comment]
Thank you!
I checked the code and found a difference between this version and fairseq version in the pre-training settings. I don't know if it is the reason that affects the pre-training results. In fairseq version, pure image and text tasks account for less. From this [line of code](https://github.com/OFA-Sys/OFA/blob/0b81ee973b9d0f5a546d62e240ed29e886c43613/data/pretrain_data/unify_dataset.py#L472),  the ratio is 8:1. ([different code of ](https://github.com/OFA-Sys/OFA-Compress/blob/a1ff4c38e922b10e509e3fe7b93a25d8a79a6126/data_utils/unify_dataset.py#L612) of Huggingface version)
Do you have any experience with the proportion of each pre-training task?


In addition, I am a little confused about some details of these three codes ([process_image_text_pair](https://github.com/OFA-Sys/OFA-Compress/blob/a1ff4c38e922b10e509e3fe7b93a25d8a79a6126/data_utils/unify_dataset.py#L268),  L12&45 of [pretrain.sh](https://github.com/OFA-Sys/OFA-Compress/blob/main/scripts/pretrain.sh#L12), and [init_task.py](https://github.com/OFA-Sys/OFA-Compress/blob/a1ff4c38e922b10e509e3fe7b93a25d8a79a6126/init_task.py#L16)). Their configurations and dimensions seem inconsistent.
